lib/rubocop/cop/rspec/scattered_setup.rb

```diff
@@ -75,8 +75,14 @@ def message(occurrences, occurrence)
         def autocorrect(corrector, first_occurrence, occurrence)
           return if first_occurrence == occurrence || !first_occurrence.body
 
+          # Heredocs are not part of the blocks source. Assume that
+          # the block consists only of code between block open/end,
+          # no funky heredoc business.
+          loc_after_open = occurrence.loc.begin.end
+          loc_after_close = occurrence.loc.end.begin
+          block_range = loc_after_open.join(loc_after_close)
```

Comment:
I’m ok with this implementation, but I cannot help but wonder if this isn’t a problem in other cops too. And if so, is this really a bug in rubocop-ast, or ast?

Reply:
But that’s why we have the MoveNode helper (and the FinalEndLocation). Haven’t looked at the issue/PR but wouldn’t they solve the problem?

Reply:
> I’m ok with this implementation, but I cannot help but wonder if this isn’t a problem in other cops too. And if so, is this really a bug in rubocop-ast, or ast?

I thought so too and opened https://github.com/rubocop/rubocop-ast/issues/293 which then contained this example:

```rb
<<~'FOO' + foo {<<~'TEXT'}
  This is FOO
FOO
    Hello World!
TEXT
```

What to me logically belongs to the block is not contiguous (I would hope this is fringe code and nobody actually writes this). After searching for cops with the same issue I couldn't actually find a single one so it didn't feel worth it to try and and something to `rubocop-ast` for this.

> But that’s why we have the MoveNode helper (and the FinalEndLocation). Haven’t looked at the issue/PR but wouldn’t they solve the problem?

I'll check that out tomorrow, would be nice if that hides this mess away. FinalEndLocation looks particularly relevant

Reply:
I couldn't make it work with `MoveNode` but the method from `FinalEndLocation` works somewhat nicely.

Reply:
Ah yes, `MoveNode` won't work here, because we are not moving the hook before or after another one, we are combining their bodies 👍